### PR TITLE
Add elasticsearch7 to devstack

### PR DESCRIPTION
**Pull request by stvstnfrd**
This looks to include some seemingly unrelated changes, presumably due to a recent rebase on my branch.

### CODE REVIEW

**Review by dianakhuang — APPROVED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -240,7 +246,7 @@ services:
     hostname: edx_notes_api.devstack.edx
     depends_on:
       - devpi
-      - elasticsearch
+      - elasticsearch7
```
Comment (regisb) [inline docker-compose.yml]: Why make only the notes service depend on elasticsearch 7, and not all services? It seems to me that either all or none of them should depend on this new service. (same comment below for the forum service)

Comment (dianakhuang) [reply docker-compose.yml]: Currently, only a few services have merged in the changes needed to support ES7. We're going to be merging the rest of them eventually, but we're trying to get these changes out a little bit at a time.

Comment (regisb) [reply docker-compose.yml]: @dianakhuang Sounds good. Would you mind documenting this somewhere, for instance in the devstack README?

**Comment by davidjoy** [comment]
So when I bring up elasticsearch7 with this PR, I see a ton of these messages in the container's logs:

```
{"type": "server", "timestamp": "2020-09-22T20:29:20,339Z", "level": "WARN", "component": "o.e.c.r.a.DiskThresholdMonitor", "cluster.name": "docker-cluster", "node.name": "elasticsearch7.devstack.edx", "message": "flood stage disk watermark [95%] exceeded on [G0QLXNgkRRmtcbgyOzKjIQ][elasticsearch7.devstack.edx][/usr/share/elasticsearch/data/nodes/0] free: 246.4mb[0.4%], all indices on this node will be marked read-only", "cluster.uuid": "vs-4go7OQCq8BKrZpvBTbg", "node.id": "G0QLXNgkRRmtcbgyOzKjIQ" }
```

I'm having issues in cs_comments_service getting it to create any new rows of data, and I'm sorta wondering if this is the cause.  Do we need to configure this thing to have more disk, perhaps?

**Comment by dianakhuang** [comment]
Seems like we might need to configure the volume to be bigger @davidjoy . The way volumes work in docker is still something like dark magic to me, so I'm hoping someone with more knowledge than me can help us figure it out.

**Comment by davidjoy** [comment]
Stand by... making sure I don't just need to prune my images.

**Comment by davidjoy** [comment]
Yup. False alarm - it was the 26GB of unused images I had on my machine. 😳

File: docker-compose.yml (hunk 1 above)
Comment (stvstnfrd) [reply docker-compose.yml]: Thanks @regisb @dianakhuang . I'm actually about to push a rebase that splits this work out into two (kinds of) pieces:
- one PR, this one, to add support for the new ES7 cluster (replacing the old ES5 one)
  - this should be merge-able now
- a series of follow-on PRs to make the switch over from old ES to the new cluster, one per "service", to be merged after/alongside the corresponding/supporting code landing in the respective service's `master` branch
  - notes (this one can merge now; notes is deployed, right?)
  - forums: to merge after `cs_comments_service` PR
  - lms/studio: to merge after `edx-search` _and_ `edx-platform` PRs

**Comment by stvstnfrd** [comment]
Just in case it got lost in the rebase-shuffle, I'll comment here again:

I just pushed a rebase that splits this work out into two (kinds of) pieces:
- one PR, this one, to add support for the new ES7 cluster (replacing the old ES5 one)
  - this should be merge-able now
- a series of follow-on PRs to make the switch over from old ES to the new cluster, one per "service", to be merged after/alongside the corresponding/supporting code landing in the respective service's `master` branch
  - notes (this one can merge now; notes is deployed, right?)
  - forums: to merge after `cs_comments_service` PR
  - lms/studio: to merge after `edx-search` _and_ `edx-platform` PRs

**Comment by regisb** [comment]
> a series of follow-on PRs to make the switch over from old ES to the new cluster, one per "service", to be merged after/alongside the corresponding/supporting code landing in the respective service's master branch 

@stvstnfrd does that mean that the devstack will be broken for the other services until then?

**Comment by dianakhuang** [comment]
@regisb devstack should continue to work for those services until the code to upgrade them to ES7 lands for real. Notes was our first one to do so, which is why it's a little more discombobulated for it.

**Comment by regisb** [comment]
Right. Assuming this fixes the CI, can you please squash your commits before we merge?

**Comment by regisb** [comment]
CI seems to be broken by another PR: https://github.com/edx/devstack/pull/620 :disappointed: 

EDIT: in fact, CI is probably broken by the fact that this branch is not rebased on top of master. @stvstnfrd @dianakhuang can you please rebase?

**Comment by stvstnfrd** [comment]
> @regisb devstack should continue to work for those services until the code to upgrade them to ES7 lands for real.

Yup, this should _not_ break existing services.
This PR adds the ability to run the new es7 container, then the subsequent "Upgrade ${service} to elasticsearch7" PRs enable es7 for the service.

**Comment by regisb** [comment]
> Yup, this should not break existing services.
> This PR adds the ability to run the new es7 container, then the subsequent "Upgrade ${service} to elasticsearch7" PRs enable es7 for the service.

Ok so I'll just merge this for now to allow you to rebase #621, #622 and #623 on top of master.

**Pull request merged**